tools/diff-processor/detector/detector.go

```diff
@@ -152,3 +161,61 @@ func suggestedTest(resourceName string, untested []string) string {
 	}
 	return strings.ReplaceAll(string(f.Bytes()), `"VALUE"`, "# value needed")
 }
+
+func DetectMissingDocs(schemaDiff diff.SchemaDiff, repoPath string) (map[string][]MissingDocField, error) {
+	missingDocFields := make(map[string][]MissingDocField)
+	for resource, resourceDiff := range schemaDiff {
+		docFilePath := resourceToDocFile(resource, repoPath)
+		var argumentsInDoc, attributesInDoc map[string]bool
+
+		content, err := os.ReadFile(docFilePath)
+		if err != nil && !os.IsNotExist(err) {
+			return nil, fmt.Errorf("failed to read resource doc %s: %w", docFilePath, err)
+		}
+		if err == nil {
+			parser := documentparser.NewParser()
+			err = parser.Parse(content)
+			if err != nil {
+				return nil, fmt.Errorf("failed to pass document %s: %w", docFilePath, err)
+			}
+			argumentsInDoc = listToMap(parser.Arguments())
+			attributesInDoc = listToMap(parser.Attributes())
+		}
+
+		for field, fieldDiff := range resourceDiff.Fields {
+			if !isNewField(fieldDiff) {
+				continue
+			}
+			// presume all fields can be arguments
```

Comment:
@shuyama1 I presume all fields can be arguments, and only some of them are attributes. Please let me know if this doesn't look right, or something else needs to be reported as missing doc. 

BTW, the high level general idea is described in the PR description.

Reply:
Thanks. As we discussed offline, we only consider top-level computed field as attributes and the rest are arguments

Reply:
sorry, the comment was not updated, i have cleaned it up now.